testing/file_test/README.md

```diff
@@ -75,18 +75,27 @@ Supported comment markers are:
 
 -   ```
     // AUTOUDPATE
+    // AUTOUDPATE:STDOUT
+    // AUTOUDPATE:STDERR
     // NOAUTOUPDATE
     ```
 
     Controls whether the checks in the file will be autoupdated if --autoupdate
-    is passed. Exactly one of these two markers must be present. If the file
-    uses splits, AUTOUPDATE must currently be before any splits.
-
-    When autoupdating, CHECKs will be inserted starting below AUTOUPDATE. When a
-    CHECK has line information, autoupdate will try to insert the CHECK
-    immediately above the line it's associated with. When that happens, any
-    following CHECK lines without line information will immediately follow,
-    between the CHECK with line information and the associated line.
+    is passed. AUTOUPDATE is equivalent to AUTOUPDATE:STDOUT followed by
+    AUTOUPDATE:STDERR. A test must specify either NOAUTOUPDATE or both
+    AUTOUPDATE markers. If the file uses splits, the AUTOUPDATE markers must
+    currently be before any splits.
+
+    When autoupdating, CHECKs will be inserted starting below the corresponding
+    AUTOUPDATE. When a CHECK has line information, autoupdate will try to insert
+    the CHECK immediately above the line it's associated with. When that
+    happens, any following CHECK lines without line information will immediately
+    follow, between the CHECK with line information and the associated line.
+
+    CHECK lines are not reordered or interleaved. If the STDOUT marker is first
+    in the file or the plain AUTOUPDATE marker is used, all STDOUT CHECKs will
+    precede all STDERR CHECKs, and if the STDERR marker is first in the file,
+    all STDERR CHECKs will precede all STDOUT CHECKs.
```

Comment:
FWIW, this still seems a bit backwards to me.

I would generally expect all STDERR output and the associated checks to *precede* STDOUT output and the associated checks as a default.

And if we do that, does it cause most (if not all?) of the cases here to work automatically?

Reply:
I tried that, but it does the wrong thing most of the time, because we put the STDOUT immediately after the STDERR, so instead of

```
// CHECK:STDOUT: lots
// CHECK:STDOUT: of
// CHECK:STDOUT: output

start of test
// CHECK:STDERR: something wrong: 'bad line'
bad line
```

we instead get

```
start of test
// CHECK:STDERR: something wrong: 'bad line'
// CHECK:STDOUT: lots
// CHECK:STDOUT: of
// CHECK:STDOUT: output
bad line
```

I also tried just forcing *all* STDOUT to the end of the file, which equally does the wrong thing in a lot of cases (when STDOUT references line numbers, it's no longer properly interleaved in the file).

Reply:
See #3073 for an attempt to follow the direction of putting STDOUT as late as possible per in-person discussion.